federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -68,16 +69,13 @@ var (
 )
 
 type ServiceController struct {
-	federationClient fedclientset.Interface
-	// A store of services, populated by the serviceController
-	serviceStore corelisters.ServiceLister
-	// Watches changes to all services
+	federationClient  fedclientset.Interface
+	serviceStore      corelisters.ServiceLister
```

Comment:
i thought they are self-explanatory.
we use `federationClient` not only for watching changes to service, but also to update them when required.